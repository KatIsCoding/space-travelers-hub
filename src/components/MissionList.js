import { useEffect } from 'react';
import { Col, Container, Row, Table } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { setMissionsAsync } from '../redux/ducks/missions';
import Mission from './Mission';

let isFirstTime = true;

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionsReducer.missions);

  useEffect(() => {
    if (isFirstTime) {
      dispatch(setMissionsAsync());
      isFirstTime = false;
    }
  }, [dispatch]);

  return (
    <Container fluid="md">
      <Row>
        <Col>
          <Table striped bordered>
            <thead>
              <tr>
                <th>Mission</th>
                <th>Description</th>
                <th>Status</th>
                <th>{}</th>
              </tr>
            </thead>
            <tbody>
              {missions.map((mission) => {
                const { mission_id: id, mission_name: name, description: desc } = mission;
                return (
                  <tr key={id}>
                    <Mission name={name} desc={desc} />
                  </tr>
                );
              })}
            </tbody>
          </Table>
        </Col>
      </Row>
    </Container>
  );
};

export default Missions;
