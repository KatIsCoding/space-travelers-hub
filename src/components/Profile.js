import { Col, Container, Row, Table } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import React from 'react';
import RocketList from './RocketsDependencies/ProfileComponent';

const Profile = () => {
  const missions = useSelector((state) => state.missionsReducer.missions);
  const myMissions = missions.filter((mission) => mission.reserved);


  return (
    <Container fluid="md">
      <Row>
        <Col xs={6}>
          <h2>My Missions</h2>
          <Table bordered>
            <tbody>
              {myMissions.map((mission) => {
                const { mission_id: id, mission_name: name } = mission;
                return (
                  <tr key={id}>
                    <td>{name}</td>
                  </tr>
                );
              })}
            </tbody>
          </Table>
        </Col>
        <Col xs={6}>
          <RocketList />
        </Col>
      </Row>
    </Container>
    
  );
};

export default Profile;
