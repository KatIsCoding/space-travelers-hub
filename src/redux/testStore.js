import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import missionsReducer from './missions/missions';
import rocketsReducer from './rockets/rockets';

const testStore = createStore(
  combineReducers({
    missionsReducer,
    rocketsReducer,
  }),
  applyMiddleware(thunk),
);

export default testStore;
