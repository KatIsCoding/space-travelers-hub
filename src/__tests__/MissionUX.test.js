import { fireEvent, render, within } from '@testing-library/react';
import fetchMock from 'fetch-mock';
import { Provider } from 'react-redux';
import MissionList from '../components/MissionList';
import Profile from '../components/Profile';
import testStore from '../redux/testStore';
import getMockMissions from '../__mocks__/getMockMissions';

let tree;
describe('testing UX for missions related components', () => {
  const url = 'https://api.spacexdata.com/v3/missions';
  fetchMock.get(url, getMockMissions());

  beforeEach(() => {
    fetchMock.reset();
    tree = render(
      <Provider store={testStore}>
        <MissionList />
        <Profile />
      </Provider>,
    );
  });

  test('testing MissionList UX', async () => {
    expect(await tree.findAllByTestId('mission')).toHaveLength(10);
    const buttons = await tree.findAllByText('Join Mission');
    fireEvent.click(buttons[0]);
    expect(await tree.findByText('Leave Mission')).toBeTruthy();
    expect(await tree.findByText('Active Member')).toBeTruthy();
    fireEvent.click(buttons[0]);
    expect(tree.queryByText('Leave Mission')).toBeFalsy();
    expect(tree.queryByText('Active Member')).toBeFalsy();
  });

  test('testing Profile UX', async () => {
    const buttons = await tree.findAllByText('Join Mission');
    fireEvent.click(buttons[1]);
    fireEvent.click(buttons[3]);
    fireEvent.click(buttons[5]);
    fireEvent.click(buttons[7]);
    fireEvent.click(buttons[9]);
    expect(await tree.findAllByTestId('profile-mission')).toHaveLength(5);
    fireEvent.click(buttons[9]);
    fireEvent.click(buttons[7]);
    expect(await tree.findAllByTestId('profile-mission')).toHaveLength(3);
    fireEvent.click(buttons[5]);
    fireEvent.click(buttons[3]);
    const profileMission = await tree.findByTestId('profile-mission');
    expect(await within(profileMission).findByText('Telstar')).toBeTruthy();
  });

  test('testing snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
