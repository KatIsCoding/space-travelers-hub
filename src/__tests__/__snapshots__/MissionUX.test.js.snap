// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UX for missions related components testing snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-md"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <table
              class="table table-striped table-bordered"
            >
              <thead>
                <tr>
                  <th>
                    Mission
                  </th>
                  <th>
                    Description
                  </th>
                  <th>
                    Status
                  </th>
                  <th />
                </tr>
              </thead>
              <tbody>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Thaicom
                  </td>
                  <td
                    class="col-6"
                  >
                    Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Telstar
                  </td>
                  <td
                    class="col-6"
                  >
                    Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-primary"
                    >
                      Active Member
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-danger"
                      type="button"
                    >
                      Leave Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Iridium NEXT
                  </td>
                  <td
                    class="col-6"
                  >
                    In 2017, Iridium began launching Iridium NEXT, a second-generation worldwide network of telecommunications satellites, consisting of 66 active satellites, with another nine in-orbit spares and six on-ground spares. These satellites will incorporate features such as data transmission that were not emphasized in the original design. The constellation will provide L-band data speeds of up to 128 kbit/s to mobile terminals, up to 1.5 Mbit/s to Iridium Pilot marine terminals, and high-speed Ka-band service of up to 8 Mbit/s to fixed/transportable terminals. The next-generation terminals and service are expected to be commercially available by the end of 2018. However, Iridium's proposed use of its next-generation satellites has raised concerns the service will harmfully interfere with GPS devices. The satellites will incorporate a secondary payload for Aireon, a space-qualified ADS-B data receiver. This is for use by air traffic control and, via FlightAware, for use by airlines. A tertiary payload on 58 satellites is a marine AIS ship-tracker receiver, for Canadian company exactEarth Ltd. Iridium can also be used to provide a data link to other satellites in space, enabling command and control of other space assets regardless of the position of ground stations and gateways.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Commercial Resupply Services
                  </td>
                  <td
                    class="col-6"
                  >
                    Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008–2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    SES
                  </td>
                  <td
                    class="col-6"
                  >
                    SES S.A. is a communications satellite owner and operator providing video and data connectivity worldwide to broadcasters, content and internet service providers, mobile and fixed network operators, governments and institutions, with a mission to “connect, enable, and enrich”. SES operates more than 50 geostationary orbit satellites and 16 medium Earth orbit satellites. These comprise the well-known European Astra TV satellites, the O3b data satellites and others with names including AMC, Ciel, NSS, Quetzsat, YahSat and SES.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    JCSAT
                  </td>
                  <td
                    class="col-6"
                  >
                    The JSAT constellation is a communication and broadcasting satellite constellation formerly operated by JSAT Corporation and currently by SKY Perfect JSAT Group. It has become the most important commercial constellation in Japan, and the fifth of the world. It has practically amalgamated all private satellite operators in Japan, with only B-SAT left as a local competitor.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    AsiaSat
                  </td>
                  <td
                    class="col-6"
                  >
                    Asia Satellite Telecommunications Holdings Limited known as its brand name AsiaSat is a commercial operator of communication spacecraft. AsiaSat is based in Hong Kong but incorporated in Bermuda.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Orbcomm OG2
                  </td>
                  <td
                    class="col-6"
                  >
                    Orbcomm Generation 2 (OG2) second-generation satellites are intended to supplement and eventually replace the current first generation constellation. Eighteen satellites were ordered by 2008—nominally intended to be launched in three groups of six during 2010–2014—and by 2015 have all been launched, on three flights. Orbcomm has options for a further thirty OG2 spacecraft. The satellites were launched by SpaceX on the Falcon 9 launch system. Originally, they were to launch on the smaller Falcon 1e rocket.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    ABS
                  </td>
                  <td
                    class="col-6"
                  >
                    ABS, formerly Asia Broadcast Satellite, is a global satellite operator based in Hong Kong and officially incorporated in Bermuda. Its services include direct-to-home and satellite-to-cable TV distribution, cellular services, and internet services. Operating 6 communication satellites, the satellite fleet currently covers 93% of the world’s population including the Americas, Africa, Asia Pacific, Europe, the Middle East, Russia and Commonwealth of Independent States.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
                <tr
                  data-testid="mission"
                >
                  <td
                    class="col-2 fw-bold"
                  >
                    Eutelsat
                  </td>
                  <td
                    class="col-6"
                  >
                    Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.
                  </td>
                  <td
                    class="col-2"
                  >
                    <span
                      class="badge bg-secondary"
                    >
                      NOT A MEMBER
                    </span>
                  </td>
                  <td
                    class="col-2"
                  >
                    <button
                      class="btn btn-outline-secondary"
                      type="button"
                    >
                      Join Mission
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="container-md"
      >
        <div
          class="row"
        >
          <div
            class="col-6"
          >
            <h2>
              My Missions
            </h2>
            <table
              class="table table-bordered"
            >
              <tbody>
                <tr
                  data-testid="profile-mission"
                >
                  <td>
                    Telstar
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="col-6"
          >
            <div
              class="list-group"
            >
              <h2>
                Rockets
              </h2>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-md"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <table
            class="table table-striped table-bordered"
          >
            <thead>
              <tr>
                <th>
                  Mission
                </th>
                <th>
                  Description
                </th>
                <th>
                  Status
                </th>
                <th />
              </tr>
            </thead>
            <tbody>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Thaicom
                </td>
                <td
                  class="col-6"
                >
                  Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Telstar
                </td>
                <td
                  class="col-6"
                >
                  Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-primary"
                  >
                    Active Member
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-danger"
                    type="button"
                  >
                    Leave Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Iridium NEXT
                </td>
                <td
                  class="col-6"
                >
                  In 2017, Iridium began launching Iridium NEXT, a second-generation worldwide network of telecommunications satellites, consisting of 66 active satellites, with another nine in-orbit spares and six on-ground spares. These satellites will incorporate features such as data transmission that were not emphasized in the original design. The constellation will provide L-band data speeds of up to 128 kbit/s to mobile terminals, up to 1.5 Mbit/s to Iridium Pilot marine terminals, and high-speed Ka-band service of up to 8 Mbit/s to fixed/transportable terminals. The next-generation terminals and service are expected to be commercially available by the end of 2018. However, Iridium's proposed use of its next-generation satellites has raised concerns the service will harmfully interfere with GPS devices. The satellites will incorporate a secondary payload for Aireon, a space-qualified ADS-B data receiver. This is for use by air traffic control and, via FlightAware, for use by airlines. A tertiary payload on 58 satellites is a marine AIS ship-tracker receiver, for Canadian company exactEarth Ltd. Iridium can also be used to provide a data link to other satellites in space, enabling command and control of other space assets regardless of the position of ground stations and gateways.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Commercial Resupply Services
                </td>
                <td
                  class="col-6"
                >
                  Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008–2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  SES
                </td>
                <td
                  class="col-6"
                >
                  SES S.A. is a communications satellite owner and operator providing video and data connectivity worldwide to broadcasters, content and internet service providers, mobile and fixed network operators, governments and institutions, with a mission to “connect, enable, and enrich”. SES operates more than 50 geostationary orbit satellites and 16 medium Earth orbit satellites. These comprise the well-known European Astra TV satellites, the O3b data satellites and others with names including AMC, Ciel, NSS, Quetzsat, YahSat and SES.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  JCSAT
                </td>
                <td
                  class="col-6"
                >
                  The JSAT constellation is a communication and broadcasting satellite constellation formerly operated by JSAT Corporation and currently by SKY Perfect JSAT Group. It has become the most important commercial constellation in Japan, and the fifth of the world. It has practically amalgamated all private satellite operators in Japan, with only B-SAT left as a local competitor.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  AsiaSat
                </td>
                <td
                  class="col-6"
                >
                  Asia Satellite Telecommunications Holdings Limited known as its brand name AsiaSat is a commercial operator of communication spacecraft. AsiaSat is based in Hong Kong but incorporated in Bermuda.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Orbcomm OG2
                </td>
                <td
                  class="col-6"
                >
                  Orbcomm Generation 2 (OG2) second-generation satellites are intended to supplement and eventually replace the current first generation constellation. Eighteen satellites were ordered by 2008—nominally intended to be launched in three groups of six during 2010–2014—and by 2015 have all been launched, on three flights. Orbcomm has options for a further thirty OG2 spacecraft. The satellites were launched by SpaceX on the Falcon 9 launch system. Originally, they were to launch on the smaller Falcon 1e rocket.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  ABS
                </td>
                <td
                  class="col-6"
                >
                  ABS, formerly Asia Broadcast Satellite, is a global satellite operator based in Hong Kong and officially incorporated in Bermuda. Its services include direct-to-home and satellite-to-cable TV distribution, cellular services, and internet services. Operating 6 communication satellites, the satellite fleet currently covers 93% of the world’s population including the Americas, Africa, Asia Pacific, Europe, the Middle East, Russia and Commonwealth of Independent States.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
              <tr
                data-testid="mission"
              >
                <td
                  class="col-2 fw-bold"
                >
                  Eutelsat
                </td>
                <td
                  class="col-6"
                >
                  Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.
                </td>
                <td
                  class="col-2"
                >
                  <span
                    class="badge bg-secondary"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="col-2"
                >
                  <button
                    class="btn btn-outline-secondary"
                    type="button"
                  >
                    Join Mission
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div
      class="container-md"
    >
      <div
        class="row"
      >
        <div
          class="col-6"
        >
          <h2>
            My Missions
          </h2>
          <table
            class="table table-bordered"
          >
            <tbody>
              <tr
                data-testid="profile-mission"
              >
                <td>
                  Telstar
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="col-6"
        >
          <div
            class="list-group"
          >
            <h2>
              Rockets
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
